test/util/codecs.test.js

```diff
@@ -0,0 +1,350 @@
+import QUnit from 'qunit';
+import {
+  codecsForPlaylist,
+  mapLegacyAvcCodecs,
+  translateLegacyCodecs
+} from '../../src/util/codecs';
+
+const generateMedia = function(isMaat, isMuxed, hasVideoCodec, hasAudioCodec, isFMP4) {
+  const codec = (hasVideoCodec ? 'avc1.deadbeef' : '') +
+    (hasVideoCodec && hasAudioCodec ? ',' : '') +
+    (hasAudioCodec ? 'mp4a.40.E' : '');
+  const master = {
+    mediaGroups: {},
+    playlists: []
+  };
+  const media = {
+    attributes: {}
+  };
+
+  if (isMaat) {
+    master.mediaGroups.AUDIO = {
+      test: {
+        demuxed: {
+          uri: 'foo.bar'
+        }
+      }
+    };
+
+    if (isMuxed) {
+      master.mediaGroups.AUDIO.test.muxed = {};
+    }
+    media.attributes.AUDIO = 'test';
+  }
+
+  if (isFMP4) {
+    // This is not a great way to signal that the playlist is fmp4 but
+    // this is how we currently detect it in HLS so let's emulate it here
+    media.segments = [
+      {
+        map: 'test'
+      }
+    ];
+  }
+
+  if (hasVideoCodec || hasAudioCodec) {
+    media.attributes.CODECS = codec;
+  }
+
+  return [master, media];
+};
+
+QUnit.module('Codec to MIME Type Conversion');
+
+const testMimeTypes = function(assert, isFMP4) {
+  // no MAAT
+  assert.deepEqual(
+    codecsForPlaylist.apply(null, generateMedia(false, true, false, false, isFMP4)),
+    {
+      audio: 'mp4a.40.2',
+      video: 'avc1.4d400d'
+    },
+    'no MAAT, codecs: none');
+
+  assert.deepEqual(
+    codecsForPlaylist.apply(null, generateMedia(false, true, true, false, isFMP4)),
+    {
+      video: 'avc1.deadbeef'
+    },
+    'no MAAT, codecs: video');
+
+  assert.deepEqual(
+    codecsForPlaylist.apply(null, generateMedia(false, true, false, true, isFMP4)),
+    {
+      audio: 'mp4a.40.E'
+    },
+    'no MAAT, codecs: audio');
+
+  assert.deepEqual(
+    codecsForPlaylist.apply(null, generateMedia(false, true, true, true, isFMP4)),
+    {
+      audio: 'mp4a.40.E',
+      video: 'avc1.deadbeef'
+    },
+    'no MAAT, codecs: video, audio');
+
+  // MAAT, not muxed
+  assert.deepEqual(
+    codecsForPlaylist.apply(null, generateMedia(true, false, false, false, isFMP4)),
+    {
+      audio: 'mp4a.40.2',
+      video: 'avc1.4d400d'
+    },
+    'MAAT, demuxed, codecs: none');
+
+  assert.deepEqual(
+    codecsForPlaylist.apply(null, generateMedia(true, false, true, false, isFMP4)),
+    {
+      audio: 'mp4a.40.2',
```

Comment:
Since it is MAAT, there should be audio. But only the video codec is included in the manifest, so we have to ensure that the audio codec is also added (we use a default one).